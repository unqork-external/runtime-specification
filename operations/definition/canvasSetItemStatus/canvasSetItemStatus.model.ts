import { DiscriminatorValue, Required } from '@tsed/schema'

import { CanvasSetItemStatusOperationOptions } from './canvasSetItemStatus.options'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.CANVAS_SET_ITEM_STATUS)
export class CanvasSetItemStatusOperation extends Operation<OperationTypes.CANVAS_SET_ITEM_STATUS> {
  @Required()
  public type = OperationTypes.CANVAS_SET_ITEM_STATUS as const

  @Required()
  public options: CanvasSetItemStatusOperationOptions = new CanvasSetItemStatusOperationOptions()
}
