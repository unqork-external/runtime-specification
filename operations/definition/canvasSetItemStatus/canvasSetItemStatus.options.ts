import { Description, Example, Optional, Required } from '@tsed/schema'

import { CanvasItemStatus } from '../../../components'

// TODO - update to TargetedOperationOptions once â€ hat PR is merged
export class CanvasSetItemStatusOperationOptions {
  @Description('Target key of the canvas component that holds this item')
  @Required()
  targetKey: string

  @Description('ID of the item to set this status on')
  @Example('f7bb5ebd-fcea-49f9-9cff-44fe81b51152')
  @Optional()
  itemId?: string

  @Description('Component key for the item to set this status on')
  @Example('myPanel')
  @Optional()
  itemKey?: string

  @Description('Status object to set or null to clear status')
  @Required()
  status: CanvasItemStatus | null
}
