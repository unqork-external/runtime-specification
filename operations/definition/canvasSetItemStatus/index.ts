export * from './canvasSetItemStatus.model'
export * from './canvasSetItemStatus.options'
