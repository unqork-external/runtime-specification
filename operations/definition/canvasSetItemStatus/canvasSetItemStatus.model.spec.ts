import { CanvasSetItemStatusOperation } from './canvasSetItemStatus.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: CanvasSetItemStatus Tests', () => {
  it('CanvasSetItemStatus Model', () => {
    const { schema } = generateSchemaAndValidate(CanvasSetItemStatusOperation)

    expect(schema).toMatchSnapshot()
  })
})
