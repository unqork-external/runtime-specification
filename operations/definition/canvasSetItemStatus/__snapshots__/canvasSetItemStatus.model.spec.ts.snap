// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: CanvasSetItemStatus Tests CanvasSetItemStatus Model 1`] = `
{
  "definitions": {
    "CanvasSetItemStatusOperationOptions": {
      "properties": {
        "itemId": {
          "description": "ID of the item to set this status on",
          "examples": [
            "f7bb5ebd-fcea-49f9-9cff-44fe81b51152",
          ],
          "type": "string",
        },
        "itemKey": {
          "description": "Component key for the item to set this status on",
          "examples": [
            "myPanel",
          ],
          "type": "string",
        },
        "status": {
          "description": "Status object to set or null to clear status",
          "type": "object",
        },
        "targetKey": {
          "description": "Target key of the canvas component that holds this item",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
        "status",
      ],
      "type": "object",
    },
  },
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/CanvasSetItemStatusOperationOptions",
    },
    "type": {
      "const": "CANVAS_SET_ITEM_STATUS",
      "examples": [
        "CANVAS_SET_ITEM_STATUS",
      ],
      "type": "object",
    },
  },
  "required": [
    "type",
    "options",
  ],
  "type": "object",
}
`;
